File: 230410-dojo/javascript/solution.js
export default function solution(n, m, section) {
  let answer = 0;
  let max = 0;

  section.forEach((wallNumber) => {
    if (wallNumber > max) {
      answer += 1;
      max = wallNumber + m - 1;
    }
  });

  return answer;
}

File: 230410-dojo/java/app/src/main/java/Solution.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int wallNumber = 0;

        for (int j : section) {
            if (j > wallNumber) {
                answer += 1;
                wallNumber = j + m - 1;
            }
        }
        return answer;
    }
}
